### PR TITLE
test: add mock test for system operations on client side

### PR DESCRIPTION
**Pull request by ZouRui89**
Signed-off-by: Zou Rui <21751189@zju.edu.cn>

### Ⅰ. Describe what this PR did
add all the mock for system operations on client side

### CODE REVIEW

**Comment by ZouRui89** [comment]
PTAL @allencloud

File: client/system_version_test.go [hunk 1]
```diff
@@ -30,11 +30,11 @@ func TestSystemVersion(t *testing.T) {
 		if !strings.HasPrefix(req.URL.Path, expectedURL) {
 			return nil, fmt.Errorf("Expected URL '%s', got '%s'", expectedURL, req.URL)
 		}
-		version := types.SystemVersion{
+		versionConfig := types.SystemVersion{
```
Comment (allencloud) [inline client/system_version_test.go]: I do not think this is a config, version is just OK for me.

File: client/registry_login.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package client
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/alibaba/pouch/apis/types"
+)
+
+// RegistryLogin requests a registry server to login.
```
Comment (allencloud) [inline client/registry_login.go]: This comment is quite confusing. Could we change this?

File: client/registry_login_test.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+func TestRegistryLogin(t *testing.T) {
+	expectedURL := "/auth"
+
+	httpClient := newMockClient(func(req *http.Request) (*http.Response, error) {
+		if !strings.HasPrefix(req.URL.Path, expectedURL) {
+			return nil, fmt.Errorf("Expected URL '%s', got '%s'", expectedURL, req.URL)
+		}
+		loginConfig := types.AuthConfig{
```
Comment (allencloud) [inline client/registry_login_test.go]: I see that in the test you input AuthConfig in `client.RegistryLogin(context.Background(), &types.AuthConfig{Username: "user_name", Password: "123456"})`. Why do you construct a new one in this place?

And actually this function would return a struct named `AuthResponse`, which is defined : 
```
type AuthResponse struct {

	// An opaque token used to authenticate a user after a successful login
	IdentityToken string `json:"IdentityToken,omitempty"`

	// The status of the authentication
	// Required: true
	Status string `json:"Status"`
}
```

Please change this part. Thanks a lot.

File: client/system_version_test.go (hunk 1 above)
Comment (ZouRui89) [reply client/system_version_test.go]: The config stuff is only for post request, right?

File: client/registry_login.go (hunk 2 above)
Comment (ZouRui89) [reply client/registry_login.go]: Sure.

File: client/registry_login_test.go (hunk 3 above)
Comment (ZouRui89) [reply client/registry_login_test.go]: I see. I mixed the get request with post request again. Will change it ASAP.

File: client/system_version_test.go (hunk 1 above)
Comment (allencloud) [reply client/system_version_test.go]: Yes, config is usually stored in the post request body. While this part of code is simulating a server which is returning a http response. And for this API, it returns `types.SystemVersion`.

File: client/registry_login.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	"context"
+	"net/http"
+
+	"github.com/alibaba/pouch/apis/types"
+)
+
+// RegistryLogin authenticates the server with a given registry to login.
+func (client *APIClient) RegistryLogin(ctx context.Context, auth *types.AuthConfig) (*types.AuthResponse, error) {
+	resp, err := client.post(ctx, "/auth", nil, auth, nil)
+	if err != nil || resp.StatusCode == http.StatusUnauthorized {
```
Comment (allencloud) [inline client/registry_login.go]: This part has a potential logic bug which is already described in https://github.com/alibaba/pouch/issues/1030.

File: client/system_info_test.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+			StatusCode: http.StatusOK,
+			Body:       ioutil.NopCloser(bytes.NewReader([]byte(b))),
+		}, nil
+	})
+
+	client := &APIClient{
+		HTTPCli: httpClient,
+	}
+
+	if infoRes, err := client.SystemInfo(context.Background()); err != nil {
```
Comment (allencloud) [inline client/system_info_test.go]: Why do we assert values when `err != nil`? @ZouRui89

**Comment by allencloud** [comment]
not-LGTM, unless update the err judging.

**Comment by allencloud** [comment]
LGTM

**Pull request merged**